### DISCUSSION TITLE
How do I migrate a legacy Orchard CMS (blog) to Orchard Core?

### DISCUSSION THREAD

**Discussion by redwyre** [OrchardCMS/OrchardCore#12655] [Q&A] [answered]
I have created a new instance of orchard core (using docker), and exported my old site, but there doesn't seem to be any way to import the legacy xml files.

**Comment by hishamco** [comment]
Orchard Core is a rewrite of Orchard CMS, OC is using to import and export

You may need to write a custom tool or a module for doing that

**Comment by ns8482e** [comment]
There is nothing ootb. As There is no one to one mapping for content type or part, for example Layout part doesn’t exists in OC 

However once you know mappings of  your definition, you can write converter that converts xml to json recipe

**Comment by redwyre** [comment, marked as answer]
My main concern is just content, layouts will need to be redone anyway. 
What I ended up doing:
* Uploaded all the media folders
* Created each blog
* Created the blog posts and just copy title and content, re-adding the images from the imported media (I originally copied the urls as well but I think I'll go back and reset them all to the new defaults for consistency)
* Export all the blogs and blog content
* Adjust the Created/published/modified dates to be the date of the original post
* importing the json to update the dates

It does seem like blogs and their content is pretty 1:1 so a best effort import should be able to get a lot of stuff across, I'd consider writing something but by the time I get proficient enough at OC the import will be a distant memory :P
